tools/topology/topology2/include/components/aec.conf

```diff
@@ -0,0 +1,160 @@
+Define {
+	AEC_UUID	"a6:a0:80:b7:9f:26:6f:46:b4:77:23:df:a0:5a:f7:58"
+}
+
+Class.Widget."aec" {
+	#
+	# Pipeline ID for the aec widget object
+	#
+	DefineAttribute."index" {}
+
+	#
+	# aec object instance
+	#
+	DefineAttribute."instance" {}
+
+	#include common component definition
+	<include/components/widget-common.conf>
+
+	DefineAttribute."num_audio_formats" {
+		# Token set reference name and type
+		token_ref	"sof_tkn_comp.word"
+	}
+
+	#
+	# aec widget UUID
+	#
+	DefineAttribute."uuid" {
+		type	"string"
+		# Token set reference name and type
+		token_ref	"sof_tkn_comp.uuid"
+	}
+
+	DefineAttribute."cpc" {
+		# Token set reference name and type
+		token_ref	"sof_tkn_comp.word"
+	}
+
+	DefineAttribute."is_pages" {
+		# Token set reference name and type
+		token_ref	"sof_tkn_comp.word"
+	}
+
+	# Attribute categories
+	attributes {
+		#
+		# The aec widget name would be constructed using the index and instance attributes.
+		# For ex: "aec.1.1" or "aec.10.2" etc.
+		#
+		!constructor [
+			"index"
+			"instance"
+		]
+
+		#
+		# mandatory attributes that must be provided when the class is instantiated
+		#
+		!mandatory [
+			"no_pm"
+			"uuid"
+			"process_type"
+		]
+
+		#
+		# immutable attributes cannot be modified in the object instance
+		#
+		!immutable [
+			"uuid"
+			"type"
+		]
+
+		#
+		# deprecated attributes should not be added in the object instance
+		#
+		!deprecated [
+			"preload_count"
+		]
+
+		#
+		# aec widget objects instantiated within the same alsaconf node must have unique
+		# instance attribute
+		#
+		unique	"instance"
+	}
+
+	num_audio_formats 2
+	num_input_audio_formats 4
+	num_output_audio_formats 2
+	Object.Base.input_audio_format {
+		1 {
+			pin_index               0
+		}
+		2 {
+			pin_index               1
+		}
+		3 {
+			pin_index               0
+			in_channels             4
+			out_channels            4
+			in_ch_cfg       $CHANNEL_CONFIG_QUATRO
+			in_ch_map       $CHANNEL_MAP_3_POINT_1
+			out_ch_cfg      $CHANNEL_CONFIG_QUATRO
+			out_ch_map      $CHANNEL_MAP_3_POINT_1
+		}
+		4 {
+			pin_index               1
+			in_channels             4
+			out_channels            4
+			in_ch_cfg       $CHANNEL_CONFIG_QUATRO
+			in_ch_map       $CHANNEL_MAP_3_POINT_1
+			out_ch_cfg      $CHANNEL_CONFIG_QUATRO
+			out_ch_map      $CHANNEL_MAP_3_POINT_1
+		}
+	}
+
+	Object.Base.output_audio_format {
+		1 {
+			in_bit_depth            16
+			in_valid_bit_depth      16
+			out_bit_depth           16
+			out_valid_bit_depth     16
+		}
+		2 {
+			in_channels             4
+			in_bit_depth            16
+			in_valid_bit_depth      16
+			out_channels            4
+			out_bit_depth           16
+			out_valid_bit_depth     16
+			in_ch_cfg       $CHANNEL_CONFIG_QUATRO
+			in_ch_map       $CHANNEL_MAP_3_POINT_1
+			out_ch_cfg      $CHANNEL_CONFIG_QUATRO
+			out_ch_map      $CHANNEL_MAP_3_POINT_1
+		}
+	}
+
+	# Default attribute values for aec widget
```

Comment:
which type?

Reply:
We can define different UUID for different solution. Currently it is for google rtc aec.

Reply:
it's not a question of UUIDs only, all the formats and capabilities will likely be different between AlgorithmA and AlgorithmZ. I don't see what kind of generic solution you can expect. This is really google-rtc-aec.conf.

Reply:
sure, thanks